Call to undefined method PHPStan\PhpDocParser\Ast\PhpDoc\PhpDocNode::getParamImmediatelyInvokedCallableTagValues() is back

### Description

I'm upgrading a Craft 4 site to Craft 5 and following [their instructions for upgrading a custom plugin](https://craftcms.com/docs/5.x/extend/updating-plugins.html#rector).

When running `vendor/bin/rector process [directory] --config vendor/craftcms/rector/sets/craft-cms-50.php` the process seems to run okay (the progress bar appears). Then I get this error multiple times (different file names) and no changes are made.

```
[ERROR] Could not process "plugins/[x]/src/variables/[x].php" file, due to:  
         "Call to undefined method                                                                        
         PHPStan\PhpDocParser\Ast\PhpDoc\PhpDocNode::getParamImmediatelyInvokedCallableTagValues()". On   
         line: 307  
```

### Steps to reproduce

1. Install as per Craft instructions 
2. Run the command above.

### Additional info

- Craft version: 4.9.3
- PHP version: 8.2.3

The version of the this I'm using is `dev-main`

--- a/composer.json
+++ b/composer.json
@@ -2,7 +2,7 @@
     "name": "craftcms/rector",
     "description": "Rector sets to automate Craft CMS upgrades",
     "require": {
-        "rector/rector": "^0.19.2"
+        "rector/rector": "^1.0.0"
     },
     "require-dev": {
         "phpunit/phpunit": "^9.0",

